src/sage/rings/function_field/drinfeld_modules/finite_drinfeld_module.py

```diff
@@ -231,39 +235,183 @@ def frobenius_charpoly(self, var='X'):
 
         ::
 
-            sage: trace = phi.frobenius_trace()
-            sage: trace
-            (4*z3^2 + 6*z3 + 3)*T + 3*z3^2 + z3 + 4
-            sage: norm = phi.frobenius_norm()
-            sage: norm
-            (5*z3^2 + 2*z3)*T^2 + (4*z3^2 + 3*z3)*T + 5*z3^2 + 2*z3
+            sage: phi.frobenius_charpoly(algorithm="NotImplemented")
+            Traceback (most recent call last):
+            NotImplementedError: Algorithm "NotImplemented" not implemented
 
-        ::
+        ALGORITHM:
 
-            sage: n = 2  # Degree of the base field over Fq
-            sage: trace.degree() <= n/2
-            True
-            sage: norm.degree() == n
-            True
+            By default, this method uses the *crystalline* algorithm which
+            computes the characteristic polynomial of the Frobenius acting
+            on the crystalline cohomology of the Drinfeld module. For further
+            details, see [Ang1997]_. Currently, the only alternative is to use
+            the *gekeler* approach based on solving the linear system given by
+            `t^{nr} + \sum_{i=1}^{r} \phi_{A_{i}}t^{n(r-i)} = 0`. For more
+            details, see [Gek2008]_.
```

Comment:
Added "so-called" and unindented the block (see https://doc.sagemath.org/html/en/developer/coding_basics.html)!

```suggestion
        By default, this method uses the so-called *crystalline*
        algorithm which computes the characteristic polynomial of the
        Frobenius acting on the crystalline cohomology of the Drinfeld
        module. For further details, see [Ang1997]_. Currently, the only
        alternative is to use the *gekeler* approach based on solving
        the linear system given by `t^{nr} + \sum_{i=1}^{r}
        \phi_{A_{i}}t^{n(r-i)} = 0`. For more details, see [Gek2008]_.t
```

When your paper is accepted, you should add it as a reference here.